src/components/Dashboard/Sidebar.tsx

```diff
@@ -0,0 +1,164 @@
+import { NavLink } from 'react-router-dom';
+import React, { useState } from 'react';
+import { motion } from 'framer-motion';
+import {
+  FaSignOutAlt, FaTable, FaOutdent, FaTh, FaRegSun,
+  FaRocketchat, FaBullhorn, FaRegUser, FaMoon, FaShoppingCart, FaHouseUser,
+} from 'react-icons/fa';
+import { MdOutlineApps } from 'react-icons/md';
+import SubMenu from './Dropdownmen';
+import { ThemeToggle } from '../navbar/ThemeToggle';
+
+function Sidebar() {
+  const sidebarAnimation = {
+    open: {
+      width: '15rem',
+      transition: {
+        damping: 40,
+      },
+    },
+    closed: {
+      display: 'none',
+      transition: {
+        damping: 40,
+      },
+    },
+  };
+
+  const [isOpen, setOpen] = useState(false);
+
+  const subMenuList = [
+    {
+      name: 'Product',
+      icon: FaTable,
+      menus: ['Add Product', 'Product List'],
+    },
+    {
+      name: 'Orders',
+      icon: FaShoppingCart,
+      menus: ['All Orders', 'Concered Orders'],
+    },
+    {
+      name: 'Users',
+      icon: FaHouseUser,
+      menus: ['All users', 'Buyer', 'Seller', 'Dissabled user'],
+    },
+  ];
+  const datadown = [{
+    icon: FaRegSun,
+    name: 'Settings',
+  },
+  {
+    icon: FaRegUser,
+    name: 'Account',
+  },
+  {
+    icon: FaSignOutAlt,
+    name: 'Logout',
+  },
+  {
+    icon: FaMoon,
+    name: 'switch mode',
+  },
+
+  ];
+  return (
+
+    <div>
+      {!isOpen && (
+        <motion.div
+          onClick={() => setOpen(true)}
+          className="absolute w-fit h-fit z-50 left-0 top-19 cursor-pointer "
+        >
+          { /* for close and open botton */}
+          <MdOutlineApps size={30} className="fill-cyan-500" />
+        </motion.div>
+      )}
+
+      <motion.div
+        variants={sidebarAnimation}
```

Comment:
Also, the ease-in animation is nice, but when close the menu, it is so robotic and abrupt. Try adding a smooth animation for closing the menu.

Reply:
that is becouse we are displaying  none.